dts/bindings/dma/st,stm32-dma.yaml

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright (c) 2019, Song Qiang <songqiang1304521@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+title: STM32 DMA
+version: 0.1
+
+description: >
+    This is a representation of the STM32 DMA nodes
+
+inherits:
+  !include dma.yaml
+
+properties:
+    compatible:
+      constraint: "st,stm32-dma"
+
+    reg:
+      type: array
+      description: register base address
+      generation: define
+      category: required
+
+    interrupts:
+      type: array
+      category: required
+      description: required irq number and irq priorities
+      generation: define
+
+    support-m2m:
+      type: bool
+      category: required
+      description: If the device supports memory to memory transfer.
+      generation: define
+
+"#cells":
+  - channel
```

Comment:
This feels like overkill and a fair amount of what should be specified in the API and not in .dts.

Reply:
Sorry I don't quit get your point. 
Do you mean I shouldn't pass parameters in dts? These parameters are passed through dts to the dma consumer driver like i2s/adc/uart driver and they translate these parameters to the standard dma consumer API parameters.
Or you mean I should use dma API defined symbols like MEMORY_TO_MEMORY instead of STM32_DMA_MEMORY_TO_MEMORY defined in stm32_dma.h?

Reply:
So the "parameters" should be what's needed to identify the dma channel, not all the runtime aspects of the dma transfer.  The application should be defining (direction, src_data_size, dst_data_size, src_burst_length, dst_burst_length, src_addr_increment, dst_addr_increment, circular_mode).  Think of the DTS as being HW description, not runtime parameters.